src/crush/CrushWrapper.h

```diff
@@ -1577,8 +1577,19 @@ class CrushWrapper {
     crush_init_workspace(crush, work);
     crush_choose_arg_map arg_map = choose_args_get_with_fallback(
       choose_args_index);
-    int numrep = crush_do_rule(crush, rule, x, rawout, maxout, &weight[0],
```

Comment:
i think a better fix is
```diff
diff --git a/src/crush/CrushWrapper.h b/src/crush/CrushWrapper.h
index 9797054603..8e3ceee48f 100644
--- a/src/crush/CrushWrapper.h
+++ b/src/crush/CrushWrapper.h
@@ -1577,8 +1577,9 @@ public:
     crush_init_workspace(crush, work);
     crush_choose_arg_map arg_map = choose_args_get_with_fallback(
       choose_args_index);
-    int numrep = crush_do_rule(crush, rule, x, rawout, maxout, &weight[0],
-                  weight.size(), work, arg_map.args);
+    int numrep = crush_do_rule(crush, rule, x, rawout, maxout,
+                  std::data(weight), std::size(weight),
+                  work, arg_map.args);
     if (numrep < 0)
       numrep = 0;
     out.resize(numrep);
```

because 

- there is no need to pass a valid pointer to `crush_do_rule()` if the size of `weight` is zero.
- we can avoid the dangerous dereference-addressof dance for retrieving the address of the array/vector
- we can accept plain C-style array by leveraging the generic C++ helpers now.

Reply:
#32673 is posted for saving the the efforts.